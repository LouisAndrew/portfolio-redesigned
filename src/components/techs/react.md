---
templateKey: techs
title: React
name: ReactJS
icon: assets/react-1-logo-png-transparent.png
---
