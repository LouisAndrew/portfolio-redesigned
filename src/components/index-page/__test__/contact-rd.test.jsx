import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'

import { render, cleanup } from '@testing-library/react'
import '@testing-library/jest-dom'

import Contact from '../contact-rd'

describe('Contact component on landing page', () => {

        afterEach( cleanup )

        const mock = {
                headingList: [
                        { h: 'Mock' },
                        { h: 'Text' },
                ],
                contactLists: [
                        {
                                social: 'email',
                                icon: 'asset/mock.png',
                                value: 'mock@mock.com',
                        }
                ]
        }

        it('renders without crashing', () => {

                const div = document.createElement('div')
                ReactDOM.render(<Contact {...mock} />, div)
        })

        it('renders image correctly', () => {

                const { getByTestId } = render( <Contact {...mock} /> )

                const leftDiv = getByTestId('left')

                // test images left div items!
                leftDiv.childNodes.forEach( (parentNode, i) => {

                        const actualImg = getByTestId(`ctc-img-${i}`)
                        expect( actualImg ).toHaveAttribute('src', `/${mock.contactLists[i].icon}`)
                        expect( actualImg ).toHaveAttribute('alt', mock.contactLists[i].social)
                } )
        })

        it('renders contact value correctly', () => {

                const { getByTestId } = render( <Contact {...mock} /> )

                const leftDiv = getByTestId('left')

                // test images left div items!
                leftDiv.childNodes.forEach( (parentNode, i) => {

                        const actualPar = getByTestId(`ctc-p-${i}`)
                        expect( actualPar ).toHaveTextContent( mock.contactLists[i].value )
                } )
        })
        it('renders heading correctly', () => {

                const { getByTestId } = render( <Contact {...mock} /> )

                const rightDiv = getByTestId('right')

                // test heading on section
                rightDiv.childNodes.forEach( parentNode => {

                        parentNode.childNodes.forEach( (node, i) => {

                                expect( node ).toHaveTextContent( mock.headingList[i].h )
                        } )
                } )
        })
        
        it('matches snapshot', () => {

                const tree = renderer.create(<Contact {...mock} />).toJSON()
                expect(tree).toMatchSnapshot()
        })

})