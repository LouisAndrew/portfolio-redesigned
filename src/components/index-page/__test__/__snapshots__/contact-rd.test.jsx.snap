// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact component on landing page matches snapshot 1`] = `
<div
  className="sc-AxjAm gSdOcq wrap"
>
  <div
    className="sc-AxirZ rxCVo"
    data-testid="left"
  >
    <div
      className="contact"
    >
      <img
        alt="email"
        data-testid="ctc-img-0"
        src="/asset/mock.png"
      />
      <p
        data-testid="ctc-p-0"
      >
        mock@mock.com
      </p>
    </div>
  </div>
  <div
    className="sc-AxiKw fAwVwF"
    data-testid="right"
  >
    <div
      className="heading"
    >
      <h1>
        Mock
      </h1>
      <h1>
        Text
      </h1>
    </div>
  </div>
</div>
`;
